rasa-ext-vim/src/Rasa/Ext/Vim.hs

```diff
@@ -178,8 +179,19 @@ normal _ = return ()
 
 -- | Move cursors to end of the line
 endOfLine :: BufAction ()
-endOfLine = findNext "\n"
+endOfLine = do 
+  txt <- getText
+  overRanges . map $ overBoth $ coordEndOfLine txt
+  where
+    coordEndOfLine :: Y.YiString -> Coord -> Coord
+    coordEndOfLine txt (Coord row col) = Coord row maxColumn
+      where
+        maxColumn :: Int
+        maxColumn = fromMaybe col (txt ^? asLines . ix row . to Y.length)
 
 -- | Move cursors to start of the line
 startOfLine :: BufAction ()
-startOfLine = findPrev "\n"
+startOfLine = overRanges . map $ overBoth coordStartOfLine
```

Comment:
I like this 👍 

Just as an interesting tidbit: You can also do the lens method: `coordRow .~ 0` in place of `coordStartOfLine`; however your version is likely more readable for folks who are unfamiliar with lens so we can keep it like that. Just figured I'd point it out 😁